Added time shift parameter

MisterProper9000 opened the pull request:
I ran into a problem that the library doesn't have the ability to start the transition from the values of object's properties that were on the transition at the moment t, which can be difficult to calculate, especially if the transition rule is complex (for example, outInBounce).

In my case, it was needed to initialize the scales of objects that approach the player from the horizon and grow up nonlinearly and which have to be between the player and the horizon when scene appears.

So I added a `time shift` parameter to all transitions (except `dissolve`) which defines initial value of object's property when the transition starts.

A short explanation of how `time shift` parameter behaves and interacts with the `delay` parameter in transition with duration `T`:
- if `time shift` > 0 then transition begins with the value of the object's properties that would have been reached by the object by time equals `time shift` with an unmodified transition and lasts for `T - time shift` ms.
- if both are defined and both > 0 then one calculation of object's properties is performed, then it waits for `delay` ms, and then transition continues for `T - time shift` ms.
- if `time shift` < 0 then trasition behaves as if `delay` is defined and equals `|time shift|`.
- if both are defined and `delay` > 0 and `time shift` < 0 then transition waits for `delay + |time shift|` ms and then starts from the initial values of object's property and lasts for `T` ms.

These code changes will not affect any existing projects.

MisterProper9000 committed:
Added time shift parameter

Added a time shift parameter for each transition type (except dissolve): as if the transition had already been in progress for some time by the moment of its start.
Updated test script to show new behaviour.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/coronalabs/framework-transition?pullRequest=2) <br/>All committers have signed the CLA.

MisterProper9000 committed:
Added blinkContinuous function

To fix typo in blink function and not affecting existing projects a blinkContinuous function was added.

MisterProper9000 committed:
Merge branch 'coronalabs:master' into master

MisterProper9000 committed:
Merge pull request #1 from MisterProper9000/hotfix-blink

Added blinkContinuous function

MisterProper9000 committed:
Added onPause call in case of time shifted transition has delay

To provide more natural behaviour onPause call was added in case if time shifted transition has delay. onPause called after one execution of object`s parameter recalculation which performs even if time shifted transition has delay and only after that thansition waits for delay ms. So there is some kind of pause happens.
